File: app/src/main/java/com/hmmelton/kibbble/models/Filters.java
package com.hmmelton.kibbble.models;

import android.support.annotation.NonNull;

/**
 * Created by harrison on 4/9/17.
 * This class stores filters used to search for pets.
 */

public class Filters {

    // Properties
    private boolean[] mSexes;
    private boolean[] mSizes;
    private boolean[] mAges;

    public Filters(@NonNull boolean[] sexes, @NonNull boolean[] sizes, @NonNull boolean[] ages) {
        mSexes = sexes;
        mSizes = sizes;
        mAges = ages;
    }

    public boolean[] getSexes() {
        return mSexes;
    }

    public boolean[] getSizes() {
        return mSizes;
    }

    public boolean[] getAges() {
        return mAges;
    }
}

File: app/src/main/java/com/hmmelton/kibbble/models/Filters.kt
package com.hmmelton.kibbble.models

/**
 * Created by harrisonmelton on 4/29/17.
 * Instances of this class act as filters for pets.
 */
class Filters(sexes: BooleanArray, sizes: BooleanArray, ages: BooleanArray) {
    // Properties
    private var mSexes: BooleanArray = sexes
    private var mSizes: BooleanArray = sizes
    private var mAges: BooleanArray = ages

    fun getSexes(): BooleanArray {
        return mSexes
    }

    fun getSizes(): BooleanArray {
        return mSizes
    }

    fun getAges(): BooleanArray {
        return mAges
    }
}